maintenance/scripts/delete_annotations.py

```diff
@@ -19,21 +19,42 @@
 
 # Delete Annotations of a particular namespace from all Images in a Dataset
 
+import argparse
 from omero.gateway import BlitzGateway
 
-USERNAME = "username"
-PASSWORD = "password"
-conn = BlitzGateway(USERNAME, PASSWORD, host="outreach.openmicroscopy.org", port=4064)
-conn.connect()
-
-# Edit these values
-dataset_id = 4501
 ns = "omero.batch_roi_export.map_ann"
 
-dataset = conn.getObject("Dataset", dataset_id)
 
-for image in dataset.listChildren():
-    ann_ids = [a.id for a in image.listAnnotations(ns)]
-    if len(ann_ids) > 0:
-        print "Deleting %s anns..." % len(ann_ids)
-        conn.deleteObjects('Annotation', ann_ids)
+def run(username, password, dataset_id, host, port):
+
+    conn = BlitzGateway(username, password, host=host, port=port)
+    try:
+        conn.connect()
+        dataset = conn.getObject("Dataset", dataset_id)
+
+        for image in dataset.listChildren():
+            ann_ids = [a.id for a in image.listAnnotations(ns)]
+            if len(ann_ids) > 0:
+                print "Deleting %s annotations..." % len(ann_ids)
+                conn.deleteObjects('Annotation', ann_ids)
+    except Exception as exc:
+            print "Error while deleting annotations: %s" % str(exc)
+    finally:
+        conn.close()
+
+
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('username')
+    parser.add_argument('password')
+    parser.add_argument('target')
```

Comment:
Same, ```dataset_id```

Reply:
Also, if we want this to applicable to other use-cases, the ```ns``` should be an argument, since the "omero.batch_roi_export.map_ann" is only used for the corresponding script.